### ISSUE TITLE
Support for "charset" in the header "Conntent-type"

### ISSUE THREAD

**Issue by LiamHaworth** [coreos/etcd#6433]
Hey Etcd Team,

I have recently been working on the Scala client for etcd (`scala-etcd`), and I have managed to get most API calls working except for one. When I make calls it is through a library called `Spray`, this library helps a lot with building the request in an easy format.

The point I am getting at here is that the library automatically builds up the whole HTTP request without much user input, when making requests to the members API to add a new member (`PUT /v2/members`) it marshals a class into a JSON object and automatically sets the `Content-type` of the request to `application/json; charset=UTF-8`.

The request fails because the etcd server responds with `{"message":"Bad Content-Type application/json; charset=utf-8, accept application/json"}`. I feel that this is a bug because setting the `charset` in `Content-type` is valid even though it isn't required for JSON documents (https://www.w3.org/International/articles/http-charset/index).

This bug can be reproduced with the following cURL command (172.17.0.0/16 being the container running etcd server)
```sh
curl http://172.17.0.2:4001/v2/members -v -XPOST -H "Content-Type: application/json; charset=utf-8" -d '{"peerURLs": ["http://172.17.0.3:2380"]}'
```

---

etcd Version: `3.0.8`
Git SHA: `d40982f`
Go Version: `go1.6.3`
Go OS/Arch: `linux/amd64`

etcd logs:

```
etcd_1  | 2016-09-14 11:43:18.691380 I | flags: recognized and used environment variable ETCD_ADVERTISE_CLIENT_URLS=http://localhost:2379,http://localhost:4001
etcd_1  | 2016-09-14 11:43:18.691478 I | flags: recognized and used environment variable ETCD_LISTEN_CLIENT_URLS=http://0.0.0.0:2379,http://0.0.0.0:4001
etcd_1  | 2016-09-14 11:43:18.691488 I | flags: recognized and used environment variable ETCD_LISTEN_PEER_URLS=http://0.0.0.0:2380,http://0.0.0.0:7001
etcd_1  | 2016-09-14 11:43:18.691543 I | etcdmain: etcd Version: 3.0.8
etcd_1  | 2016-09-14 11:43:18.691548 I | etcdmain: Git SHA: d40982f
etcd_1  | 2016-09-14 11:43:18.691551 I | etcdmain: Go Version: go1.6.3
etcd_1  | 2016-09-14 11:43:18.691555 I | etcdmain: Go OS/Arch: linux/amd64
etcd_1  | 2016-09-14 11:43:18.691560 I | etcdmain: setting maximum number of CPUs to 8, total number of available CPUs is 8
etcd_1  | 2016-09-14 11:43:18.691564 W | etcdmain: no data-dir provided, using default data-dir ./default.etcd
etcd_1  | 2016-09-14 11:43:18.691622 I | etcdmain: listening for peers on http://0.0.0.0:2380
etcd_1  | 2016-09-14 11:43:18.691639 I | etcdmain: listening for peers on http://0.0.0.0:7001
etcd_1  | 2016-09-14 11:43:18.691663 I | etcdmain: listening for client requests on 0.0.0.0:2379
etcd_1  | 2016-09-14 11:43:18.691677 I | etcdmain: listening for client requests on 0.0.0.0:4001
etcd_1  | 2016-09-14 11:43:18.991227 I | etcdserver: name = default
etcd_1  | 2016-09-14 11:43:18.991262 I | etcdserver: data dir = default.etcd
etcd_1  | 2016-09-14 11:43:18.991277 I | etcdserver: member dir = default.etcd/member
etcd_1  | 2016-09-14 11:43:18.991287 I | etcdserver: heartbeat = 100ms
etcd_1  | 2016-09-14 11:43:18.991296 I | etcdserver: election = 1000ms
etcd_1  | 2016-09-14 11:43:18.991306 I | etcdserver: snapshot count = 10000
etcd_1  | 2016-09-14 11:43:18.991325 I | etcdserver: advertise client URLs = http://localhost:2379,http://localhost:4001
etcd_1  | 2016-09-14 11:43:18.991337 I | etcdserver: initial advertise peer URLs = http://localhost:2380
etcd_1  | 2016-09-14 11:43:18.991369 I | etcdserver: initial cluster = default=http://localhost:2380
etcd_1  | 2016-09-14 11:43:19.295857 I | etcdserver: starting member 8e9e05c52164694d in cluster cdf818194e3a8c32
etcd_1  | 2016-09-14 11:43:19.296021 I | raft: 8e9e05c52164694d became follower at term 0
etcd_1  | 2016-09-14 11:43:19.296052 I | raft: newRaft 8e9e05c52164694d [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
etcd_1  | 2016-09-14 11:43:19.296067 I | raft: 8e9e05c52164694d became follower at term 1
etcd_1  | 2016-09-14 11:43:19.817687 I | etcdserver: starting server... [version: 3.0.8, cluster version: to_be_decided]
etcd_1  | 2016-09-14 11:43:19.819182 I | membership: added member 8e9e05c52164694d [http://localhost:2380] to cluster cdf818194e3a8c32
etcd_1  | 2016-09-14 11:43:19.904631 I | raft: 8e9e05c52164694d is starting a new election at term 1
etcd_1  | 2016-09-14 11:43:19.904751 I | raft: 8e9e05c52164694d became candidate at term 2
etcd_1  | 2016-09-14 11:43:19.904780 I | raft: 8e9e05c52164694d received vote from 8e9e05c52164694d at term 2
etcd_1  | 2016-09-14 11:43:19.904827 I | raft: 8e9e05c52164694d became leader at term 2
etcd_1  | 2016-09-14 11:43:19.904881 I | raft: raft.node: 8e9e05c52164694d elected leader 8e9e05c52164694d at term 2
etcd_1  | 2016-09-14 11:43:19.948422 W | etcdserver: apply entries took too long [43.059576ms for 2 entries]
etcd_1  | 2016-09-14 11:43:19.948466 W | etcdserver: avoid queries with large range/delete range!
etcd_1  | 2016-09-14 11:43:19.948572 I | etcdserver: setting up the initial cluster version to 3.0
etcd_1  | 2016-09-14 11:43:19.948654 I | etcdmain: ready to serve client requests
etcd_1  | 2016-09-14 11:43:19.948900 I | etcdserver: published {Name:default ClientURLs:[http://localhost:2379 http://localhost:4001]} to cluster cdf818194e3a8c32
etcd_1  | 2016-09-14 11:43:19.948952 I | etcdmain: ready to serve client requests
etcd_1  | 2016-09-14 11:43:19.992521 N | membership: set the initial cluster version to 3.0
etcd_1  | 2016-09-14 11:43:19.992645 I | api: enabled capabilities for version 3.0
etcd_1  | 2016-09-14 11:43:19.995774 N | etcdmain: serving insecure client requests on 0.0.0.0:2379, this is strongly discouraged!
etcd_1  | 2016-09-14 11:43:20.036554 N | etcdmain: serving insecure client requests on 0.0.0.0:4001, this is strongly discouraged!
```

**Comment by heyitsanthony** [comment]
Seems reasonable. The content type check can probably ignore everything past ";".

**Comment by xiang90** [comment]
mark this as p2 since this is not a critical issue.

**Comment by LiamHaworth** [comment]
At P2 how long would you guys think it would be until a fix for this is released?

**Comment by xiang90** [comment]
@LiamHaworth It means this is not urgent. So we would not prioritize for the next back-port. If you want it to be fix faster, you can always help. Thanks!

**Comment by LiamHaworth** [comment]
I would love to, would just need to get my self comfortable with the code base first, we shall see.

**Comment by LiamHaworth** [comment]
Implemented a fix in the pull request #6435

**Comment by xiang90** [comment]
Fixed by #6436
